### PR TITLE
KG 12: clean up Exocom's HTTP subsystem

### PR DESCRIPTION
**Pull request by kevgo**
The HTTP subsystem had a bunch of technical drift like old names that no longer apply, and a separation into 2 files that no longer makes sense. This PR simplifies it to one file and improves the way it signals events to the Exocom core. 

@hugobho @trushton

### CODE REVIEW

File: exocom-server/src/exocom.ls [hunk 1]
```diff
@@ -15,12 +15,11 @@ class ExoCom extends EventEmitter
...
   ({@service-messages} = {}) ->
     @client-registry    = new ClientRegistry {@service-messages}
-    @listener-subsystem = new ListenerSubsystem @
+    @http-subsystem = new HttpSubsystem @
+      ..on 'online', (port) ~> @emit 'http-online', port
     @message-cache = new MessageCache!
     @websocket = new WebSocketSubsystem @
 
-    delegate 'httpPort', from: @, to: @listener-subsystem
-    delegate-event 'http-bound' 'error', from: @listener-subsystem, to: @
```
Comment (hugobho) [inline exocom-server/src/exocom.ls]: Shouldn't `error` from `@http-subsystem` still be bubbled up to `@`?

Comment (hugobho) [inline exocom-server/src/exocom.ls]: Nevermind, I see that you've made those changes in another PR.

Comment (hugobho) [inline exocom-server/src/exocom.ls]: Nevermind, I see you've made those changes in another PR.

Comment (kevgo) [inline exocom-server/src/exocom.ls]: Good catch, that change probably slipped into the wrong PR!

**Pull request closed without merging**